GVFS/GVFS/CommandLine/SparseVerb.cs

```diff
@@ -153,6 +147,81 @@ protected override void Execute(GVFSEnlistment enlistment)
             }
         }
 
+        private void CleanUpFoldersOutsideSparse(ITracer tracer, Enlistment enlistment, HashSet<string> sparseFolders)
+        {
+            string[] directoriesToDehydrate = this.GetDirectoriesOutsideSparse(enlistment.WorkingDirectoryBackingRoot, sparseFolders);
+            if (!this.ShowStatusWhileRunning(
+                () =>
+                {
+                    ReturnCode verbReturnCode = this.ExecuteGVFSVerb<DehydrateVerb>(
+                        tracer,
+                        verb =>
+                        {
+                            verb.Confirmed = true;
+                            verb.Folders = string.Join(";", directoriesToDehydrate);
+                        });
+
+                    return verbReturnCode == ReturnCode.Success;
+                },
+                "Cleaning folders"))
+            {
+                this.ReportErrorAndExit(tracer, "Failed to dehydrate.");
+            }
+        }
+
+        private string[] GetDirectoriesOutsideSparse(string rootPath, HashSet<string> sparseFolders)
+        {
+            PhysicalFileSystem fileSystem = new PhysicalFileSystem();
+            Queue<string> foldersToEnumerate = new Queue<string>();
+            foldersToEnumerate.Enqueue(rootPath);
+
+            List<string> foldersToRemove = new List<string>();

```

Comment:
This method does not actually remove folders - maybe something like `directoriesOutsideSparse` or `foldersOutsideSparse`?